src/dsp/generator.cpp

```diff
@@ -232,6 +248,261 @@ void KernelOp<InterpolationTypes::ZeroOrderHold, T>::Process(
     }
 }
 
+template <int NUM_CHANNELS, bool LOOP_ACTIVE>
+void KernelOp<InterpolationTypes::ZeroOrderHoldAA, float>::Process(
+    GeneratorState *__restrict GD,
+    KernelProcessor<InterpolationTypes::ZeroOrderHoldAA, float, NUM_CHANNELS, LOOP_ACTIVE> &ks)
+{
+    auto readSampleL{ks.ReadSample[0]};
+    auto readFadeSampleL{ks.ReadFadeSample[0]};
+    auto OutputL{ks.Output[0]};
+    auto m0{ks.m0};
+    auto i{ks.i};
+
+#if DEBUG_CODE_I_WANT_TO_RETAIN
```

Comment:
I think we just use `#if DEBUG`.

Reply:
This was in the original codebase. It's mostly copy and paste from the sinc interpolation methods, the major differece between the value of the subpos variables.

Reply:
No that's all fine and good, but it makes sense to have a generic #if here that would apply possibly to debugging other resampling modes in the future.

Reply:
Nah we want to leave it as such

if debug will always turn it on

we can rename it generator debug code to retain or something but I called it that to stop a future cleanup from removing it 

I turn it in at debug time st the top of the file if I need it

Reply:
This #if is a new addition here, though, it doesn't seem something you had beforehand under that name? Or am I reading this diff wrong.

Reply:
It’s copied from the one I wrote 

but if debug will turn it in too much so it needs removal or a different name

Reply:
Right so we agree, something like #if DEBUG_RESAMPLERS would do :)

Reply:
yup

i maned it "DEBUG I WANT TO RETAIN" so we wouldn't remove it in a cleanup sweep but indeed DEBUG_GENERATOR would be better